### ISSUE TITLE
[a11y] Toast not conveyed consistently by screen readers

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#1215]
# Issue summary

The toast component may cause issues for assistive tech users on Windows.

NVDA doesn't announce the `aria-live` container when it appears. This is related to how NVDA parses `aria-live` regions. To get the content to be read, the container with `aria-live="..."`, `role="alert"`, or `role="status"` needs to be in the DOM already when its contents are changed. This is something that doesn't seem to be likely to change.

Here's an NVDA issue about it for some more context: https://github.com/nvaccess/nvda/issues/8873

And MDN provides an overview on having the `role` or `aria-live` attribute present on page load: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions

This issue was originally discovered and reported while testing https://github.com/Shopify/polaris-react/issues/553.

Additionally, the `X` button on toasts has no button text available to screen reader users.

## Expected behavior

The toast content should be read when it appears.

## Actual behavior

NVDA with Firefox does not announce the toast.

## Steps to reproduce the problem

1. Navigate to any toast example using Firefox on Windows (such as https://polaris.shopify.com/components/feedback-indicators/toast).
1. With NVDA running, activate the control that displays the toast.

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number: 3.10.0+
- Browser: Firefox
- Device: Windows VM on Mac
- Operating System: Windows 10
